### ISSUE TITLE
A cause to Go bad performance ?

### ISSUE THREAD

**Issue by LeVraiBaptiste** [matttomasetti/Go-Gorilla_Websocket-Benchmark-Server#2]
Hello there,
I was reading your paper and was flabbergasted by go atrocious performance.
I don't know if go routines where invented three years ago but now it's a pretty standard way to parallelize work, and in a way, make it more asynchronous.
Would love to see the benchmark with a new go script where every websocket client is handled in its own go routine;
I guess the results would not be the same.
Love from France

**Comment by westofpluto** [comment]
Yes something here doesn't quite smell right. The Gorilla package is built on top of the http package which does indeed use goroutines, one for each client. Was this not the case 3 years ago? If each client does indeed run the handler in its own goroutine then there is no way that the performance of Go/Gorilla should be that bad.
